src/qibolab/instruments/qblox/cluster_qcm_bb.py

```diff
@@ -151,162 +192,31 @@ def connect(self, cluster: QbloxCluster = None):
-                self._set_device_parameter(
-                    target, "sync_en", value=False
-                )  # Default after reboot = False
-                self._set_device_parameter(
-                    target,
-                    "upsample_rate_awg_path0",
-                    "upsample_rate_awg_path1",
-                    value=0,
-                )
-
-            for target in [
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o1"]],
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o2"]],
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o3"]],
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o4"]],
-            ]:
-                self._set_device_parameter(
-                    target, "connect_out0", value="off"
-                )  # Default after reboot = True
-                self._set_device_parameter(
-                    target, "connect_out1", value="off"
-                )  # Default after reboot = True
-                self._set_device_parameter(
-                    target, "connect_out2", value="off"
-                )  # Default after reboot = True
-                self._set_device_parameter(
-                    target, "connect_out3", value="off"
-                )  # Default after reboot = True
-
-            self._set_device_parameter(
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o1"]],
-                "connect_out0",
-                value="I",
-            )
-            self._set_device_parameter(
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o2"]],
-                "connect_out1",
-                value="Q",
-            )
-            self._set_device_parameter(
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o3"]],
-                "connect_out2",
-                value="I",
-            )
-            self._set_device_parameter(
-                self.device.sequencers[self.DEFAULT_SEQUENCERS["o4"]],
-                "connect_out3",
-                value="Q",
-            )
-
-            # on initialisation, disconnect all other sequencers from the ports
-            self._device_num_sequencers = len(self.device.sequencers)
-            for sequencer in range(4, self._device_num_sequencers):
-                target = self.device.sequencers[sequencer]
-                self._set_device_parameter(target, "connect_out0", value="off")
-                self._set_device_parameter(target, "connect_out1", value="off")
-                self._set_device_parameter(target, "connect_out2", value="off")
-                self._set_device_parameter(target, "connect_out3", value="off")
+            self._set_default_values()
+            # then set the value loaded from the runcard
             try:
                 for port in self.settings:
                     self._sequencers[port] = []
                     self.ports[port].offset = self.settings[port]["offset"]
```

Comment:
I have the impression that the `self.settings` dictionary can also be dropped, but maybe this can be done in a different PR to keep the changes smaller. The value of the `offset` (or any other parameter) is already cached within the `QbloxOutputPort_Settings` object which is a more well defined data structure than the dictionary so we don't need it in a second place.

During `connect` we just need to make sure that the values stored in `Qblox*Port_Settings` are uploaded (`device.set`) to the instrument.

Reply:
Thanks @stavros11 , I think is better to do it in a following PR and merge this. I mark it as resolved and I opened a issue about this.